impeller/toolkit/interop/texture.h

```diff
@@ -0,0 +1,41 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_IMPELLER_TOOLKIT_INTEROP_TEXTURE_H_
+#define FLUTTER_IMPELLER_TOOLKIT_INTEROP_TEXTURE_H_
+
+#include "impeller/core/texture.h"
+#include "impeller/display_list/dl_image_impeller.h"
+#include "impeller/toolkit/interop/context.h"
+#include "impeller/toolkit/interop/impeller.h"
+#include "impeller/toolkit/interop/object.h"
+
+namespace impeller::interop {
+
+class Texture final
+    : public Object<Texture, IMPELLER_INTERNAL_HANDLE_NAME(ImpellerTexture)> {
+ public:
+  Texture(const Context& context, const TextureDescriptor& descriptor);
+
+  ~Texture() override;
+
+  Texture(const Texture&) = delete;
+
+  Texture& operator=(const Texture&) = delete;
+
+  bool IsValid() const;
+
+  bool SetContents(const uint8_t* contents, uint64_t length);
```

Comment:
Please do not expose either of these methods. See https://github.com/flutter/flutter/issues/144498 and https://github.com/flutter/flutter/issues/147945

We could instead expose a static method that constructs an initial texture from these bytes but does not support updating teh contents.

Reply:
Thats a good call on the static method. Updating.